src/mod_extdisco.erl

```diff
@@ -20,66 +20,70 @@
 -behaviour(gen_mod).
 
 %% gen_mod callbacks.
--export([start/2, stop/1, config_spec/0]).
+-export([start/2, stop/1, supported_features/0, config_spec/0]).
 
--export([process_iq/4]).
+-export([process_iq/5]).
 
--ignore_xref([process_iq/4]).
+-ignore_xref([process_iq/5]).
 
 -include("jlib.hrl").
--include("mongoose.hrl").
 -include("mongoose_config_spec.hrl").
 
--spec start(jid:server(), list()) -> ok.
-start(Host, Opts) ->
-    IQDisc = gen_mod:get_opt(iqdisc, Opts, no_queue),
-    gen_iq_handler:add_iq_handler(ejabberd_local, Host,
-        ?NS_EXTDISCO, ?MODULE, process_iq, IQDisc).
+-spec start(mongooseim:host_type(), gen_mod:module_opts()) -> ok.
+start(HostType, #{iqdisc := IQDisc}) ->
+    gen_iq_handler:add_iq_handler_for_domain(HostType, ?NS_EXTDISCO, ejabberd_local,
+                                             fun ?MODULE:process_iq/5, #{}, IQDisc).
 
--spec stop(jid:server()) -> ok.
-stop(Host) ->
-    gen_iq_handler:remove_iq_handler(ejabberd_local, Host, ?NS_EXTDISCO).
+-spec stop(mongooseim:host_type()) -> ok.
+stop(HostType) ->
+    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_EXTDISCO, ejabberd_local).
+
+supported_features() ->
+    [dynamic_domains].
 
 -spec config_spec() -> mongoose_config_spec:config_section().
 config_spec() ->
     #section{
-        items = #{<<"service">> => #list{items = service_config_spec(),
-                                         wrap = none}}
-        }.
+       items = #{<<"iqdisc">> => mongoose_config_spec:iqdisc(),
+                 <<"service">> => #list{items = service_config_spec()}},
+       defaults = #{<<"iqdisc">> => no_queue,
+                    <<"service">> => []},
+       format_items = map
+      }.
 
 service_config_spec() ->
     #section{
-        items = #{<<"type">> => #option{type = atom,
-                                        validate = non_empty},
-                  <<"host">> => #option{type = string,
-                                        validate = non_empty},
-                  <<"port">> => #option{type = integer,
-                                        validate = port},
-                  <<"transport">> => #option{type = string,
-                                             validate = {enum, ["udp", "tcp"]}},
-                  <<"username">> => #option{type = string,
-                                            validate = non_empty},
-                  <<"password">> => #option{type = string,
-                                            validate = non_empty}
-            },
-        required = [<<"type">>, <<"host">>]
+       items = #{<<"type">> => #option{type = atom,
```

Comment:
Minor, but I like the old indentation more.

Reply:
That's Erlang Mode for Emacs in action... I changed the syntax to avoid this strange case.